parlai/tasks/wizard_of_internet/agents.py

```diff
@@ -443,10 +452,16 @@ def teacher_setup_data(self, datafile) -> Message:
                         f' {self.gold_knowledge_delimiter} {text}'
                     ),
                 )
+                if self.add_skip_search_if_gold_prepended:
+                    message[CONST.SKIP_SEARCH] = True
             yield message, episode_started
 
 
 class WizardDialogGoldKnowledgeTeacher(WizardDialogTeacher):
+    def __init__(self, opt, shared=None):
+        super().__init__(opt, shared=shared)
+        self.id = 'WizardDialogGoldKnowledgeTeacher'
```

Comment:
ah thanks for adding this, otherwise the train_model also shows this red flag ;)